crates/nassun/src/fetch/npm.rs

```diff
@@ -145,3 +145,150 @@ impl PackageFetcher for NpmFetcher {
         Ok(self.client.stream_external(url).await?)
     }
 }
+
+#[cfg(test)]
+mod test {
+    use node_semver::{Range, Version};
+    use oro_package_spec::VersionSpec;
+    use tempfile::tempdir;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn read_name() -> Result<()> {
+        let fetcher = NpmFetcher {
+            client: oro_client::OroClient::default(),
+            registries: HashMap::default(),
+            corgi_packuments: DashMap::default(),
+            packuments: DashMap::default(),
+        };
```

Comment:
I think you should just use `NpmFetcher::new` here, tbh

Reply:
Fixed. Makes things way easier to read.